jpsonic-main/src/main/java/com/tesshu/jpsonic/service/upnp/processor/AlbumByGenreUpnpProcessor.java

```diff
@@ -22,88 +22,68 @@
-        return m;
-    }
-
-    public AlbumByGenreUpnpProcessor(@Lazy UpnpProcessDispatcher d, UpnpProcessorUtil u, MediaFileService m,
-            SearchService s) {
-        super(d, u);
-        this.util = u;
-        this.mediaFileService = m;
-        this.searchService = s;
-        setRootId(UpnpProcessDispatcher.CONTAINER_ID_ALBUM_BY_GENRE_PREFIX);
+    public AlbumByGenreUpnpProcessor(UpnpProcessorUtil util, UpnpDIDLFactory factory, MediaFileService mediaFileService,
+            SearchService searchService) {
+        super();
+        this.util = util;
+        this.factory = factory;
+        this.mediaFileService = mediaFileService;
+        this.searchService = searchService;
     }
 
-    @PostConstruct
     @Override
-    public void initTitle() {
-        setRootTitleWithResource("dlna.title.albumbygenres");
+    public ProcId getProcId() {
+        return ProcId.ALBUM_BY_GENRE;
     }
 
     @Override
-    public BrowseResult browseRoot(String filter, long firstResult, long maxResults, SortCriterion... orderBy)
-            throws ExecutionException {
+    public BrowseResult browseRoot(String filter, long firstResult, long maxResults) throws ExecutionException {
         DIDLContent didl = new DIDLContent();
         List<MediaFile> selectedItems = getItems(firstResult, maxResults);
         for (int i = 0; i < selectedItems.size(); i++) {
             MediaFile item = selectedItems.get(i);
             didl.addContainer(createContainer(item, Integer.toString((int) (i + firstResult))));
         }
-        return createBrowseResult(didl, (int) didl.getCount(), getItemCount());
+        return createBrowseResult(didl, (int) didl.getCount(), getDirectChildrenCount());
     }
 
     @Override
     public Container createContainer(MediaFile item) {
         MusicAlbum container = new MusicAlbum();
-        container.setAlbumArtURIs(new URI[] { getDispatcher().getMediaFileProcessor().createAlbumArtURI(item) });
+        container.addProperty(factory.toAlbumArt(item));
         if (item.getArtist() != null) {
-            container.setArtists(getDispatcher().getAlbumProcessor().getAlbumArtists(item.getArtist())
-                    .toArray(new PersonWithRole[0]));
+            container.addProperty(factory.toPerson(item.getArtist()));
         }
         container.setDescription(item.getComment());
-        container.setId(UpnpProcessDispatcher.CONTAINER_ID_FOLDER_PREFIX + UpnpProcessDispatcher.OBJECT_ID_SEPARATOR
-                + item.getId());
+        container.setId(ProcId.FOLDER.getValue() + ProcId.CID_SEPA + item.getId());
         container.setTitle(item.getName());
         container.setChildCount(getChildSizeOf(item));
         if (mediaFileService.isRoot(item)) {
-            container.setParentID(getRootId());
+            container.setParentID(getProcId().getValue());
         } else {
             MediaFile parent = mediaFileService.getParentOf(item);
             if (parent != null) {
```

Comment:
Add test.

Reply:
Yeah, I think this will be done when DIDL is standardized when MusicIndex is redesigned. It will be checked at that time.